### PR TITLE
feat: Configure cloudwatch logging for step functions module

### PR DESCRIPTION
**Pull request by ahummel25**
## Description
Adding `logging_configuration` property to enable Cloudwatch logging for step functions

## Motivation and Context
Need ability to utilize Cloudwatch logging for step functions

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

**Comment by ahummel25** [comment]
Looks like my IDE took some liberties reformatting the `README`. Let me try fixing that.

**Comment by ahummel25** [comment]
Actually the `README` looks completely fine in preview mode

File: README.md [hunk 1]
```diff
@@ -4,18 +4,16 @@ Terraform module, which creates AWS Step Functions as well as required IAM role
...
 - [x] Creates AWS Step Function
 - [x] Conditional creation for many types of resources
 - [x] Support IAM policy attachments for [Integrated Services (eg, Lambda, SQS, ECS, EKS, Batch, DynamoDB, etc)](https://docs.aws.amazon.com/step-functions/latest/dg/service-integration-iam-templates.html) and various ways to create and attach additional policies
 
-
 ## Usage
 
 ### Step Function
-  
+
```
Comment (antonbabenko) [inline README.md:16]: There is no need to touch README manually, run `pre-commit install` and `pre-commit run -a` to do required updates in the docs.

File: main.tf [hunk 2]
```diff
@@ -13,6 +24,15 @@ resource "aws_sfn_state_machine" "this" {
   role_arn   = var.use_existing_role ? var.role_arn : aws_iam_role.this[0].arn
   definition = var.definition
 
+  dynamic "logging_configuration" {
+    for_each = var.logging_configuration.level == "OFF" ? [] : list(var.logging_configuration)
```
Comment (antonbabenko) [inline main.tf:28]: User may want to bring another cloudwatch_log_group created outside of this module. Please make required changes (see https://github.com/terraform-aws-modules/terraform-aws-lambda module for reference).

Comment (antonbabenko) [reply main.tf:28]: `list()` => `[]`

File: variables.tf [hunk 3]
```diff
@@ -146,6 +146,57 @@ variable "attach_policies" {
   default     = false
 }
 
+
+##########################
+# Cloudwatch Logging
+##########################
+variable "logging_configuration" {
+  description = "Defines what execution history events are logged and where they are logged"
+  type = object({
```
Comment (antonbabenko) [inline variables.tf:155]: Use `map()` instead of `object()` type.

File: variables.tf [hunk 4]
```diff
@@ -146,6 +146,57 @@ variable "attach_policies" {
   default     = false
 }
 
+
+##########################
```
Comment (antonbabenko) [inline variables.tf:150]: Move all cloudwatch vars to a better place (eg, line 57). Now they are in the middle of the policies section.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Please update the code a bit.

File: main.tf (hunk 2 above)
Comment (ahummel25) [reply main.tf:28]: > User may want to bring another cloudwatch_log_group created outside of this module. Please make required changes (see https://github.com/terraform-aws-modules/terraform-aws-lambda module for reference).

Sorry, not sure I understand this. You want me to make `aws_cloudwatch_log_group` a separate module?

**Comment by ahummel25** [comment]
@antonbabenko Addressed PR comments. Let me know if I've missed the mark on anything.

**Comment by atrakic** [comment]
Looking fwd to see this PR merged.

**Comment by svenlito** [comment]
@ahummel25 Could resolve the conflict, please?

**Comment by ahummel25** [comment]
@svenlito Done.

@antonbabenko I've resolved the merge conflicts, but not sure why the Pre-Commit hook is failing. Can you please advise?

**Comment by bryantbiggs** [comment]
@ahummel25 if you update the `.github/workflows/pre-commit.yml` as shown here, your checks should pass - https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/139/files#diff-ac5eead3f3ce4863c524fff031a87b7aecccb4a0493df087a4e1c704a1505036R97

its the terraform docs that we are still unfortunately fighting with at the moment

**Comment by ahummel25** [comment]
@bryantbiggs Just did so. Looks like it's still failing. Any ideas?

**Comment by bryantbiggs** [comment]
so that fixed the pipeline but it looks like you haven't run the pre-commit hook on the codebase so you have a diff 
image

**Comment by ahummel25** [comment]
I just ran `pre-commit run -a` locally and nothing got updated. Is that the right command to run?

**Comment by bryantbiggs** [comment]
thats correct, but I suspect that your local version of `terraform-docs` is out of date

```
$ terraform-docs --version
> terraform-docs version v0.12.1 darwin/amd64 BuildDate: 2021-03-29T18:15:47+0100
```

**Comment by ahummel25** [comment]
Thanks @bryantbiggs. That fixed it.

**Comment by antonbabenko** [comment]
Thanks @ahummel25 !

[v1.3.0](https://github.com/terraform-aws-modules/terraform-aws-step-function/releases/v1.3.0) has been just released.

**Pull request merged**